applications/nrf5340_audio/src/audio/audio_datapath.c

```diff
@@ -693,69 +693,62 @@ static void audio_datapath_i2s_blk_complete(uint32_t frame_start_ts_us, uint32_t
 
 	/********** I2S RX **********/
 	static int prev_ret;
-	struct audio_data *rx_audio_block = NULL;
-	struct net_buf *buf_in;
+	struct net_buf *rx_audio_block = NULL;
 
 	if (IS_ENABLED(CONFIG_STREAM_BIDIRECTIONAL) || (CONFIG_AUDIO_DEV == GATEWAY)) {
 		if (rx_buf_released == NULL) {
 			/* No RX data available */
 			ERR_CHK_MSG(-ENOMEM, "No RX data available");
 		}
 
-		buf_in = net_buf_alloc(&pool_i2s_rx, K_NO_WAIT);
-		if (buf_in == NULL) {
-			ERR_CHK_MSG(-ENOMEM, "Out of RX buffers for I2S");
-		}
+		uint32_t q_num_free = k_msgq_num_free_get(ctrl_blk.in.audio_q);
```

Comment:
Nit: Since the queue is now called `audio_q` it would make sense to rename `q_num_free` to `audio_q_num_free`